### PR TITLE
Automatically get latest Swift tag in SAIO/SAIT

### PR DESCRIPTION
**Pull request by gerardgine**
It is now ready for review.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -198,7 +198,7 @@ pre-generate:
 
 python-test:
 	cd meta_middleware && tox -e lint
-	cd pfs_middleware && tox -e py27-release,py27-minver,py36-release,lint
+	cd pfs_middleware && LATEST_SWIFT_TAG=` git ls-remote --refs --tags https://github.com/NVIDIA/swift.git | cut -d '/' -f 3 | grep ^ss-release- | sort --version-sort | tail -n 1 ` tox -e py27-release,py27-minver,py36-release,lint
```
Comment (edmc-ss) [inline Makefile:201]: Really cool that you were able to figure out the TOX-specific change, too!

File: saio/container/Dockerfile [hunk 2]
```diff
@@ -111,7 +111,7 @@ RUN pip install -r test-requirements.txt
 WORKDIR /home/swift
 RUN git clone https://github.com/NVIDIA/swift.git
 WORKDIR /home/swift/swift
-RUN git checkout $SwiftVersion
+RUN if [[ -n "$SwiftVersion" ]] ; then git checkout $SwiftVersion ; else git checkout $( git describe --tags --abbrev=0 $(git rev-list --tags --max-count=1) ) ; fi
```
Comment (edmc-ss) [inline saio/container/Dockerfile:114]: I assume this works...

Key thing, I think, that such a derivation would necessarily affect the Image "TAG" that gets specified as the result of the "docker build" command... perhaps that would necessitate a different approach than this (i.e. the version/tag derivation would be done in a "wrapper" around "docker build" that would, instead, explicitly specify the SwiftVersion arg to the Dockerfile... and, of course, specify the target name of the built Docker Image.

File: saio/vagrant_provision.sh [hunk 3]
```diff
@@ -228,7 +228,8 @@ echo "export ST_KEY=testing" >> ~vagrant/.bash_profile
 cd ~swift
 git clone https://github.com/NVIDIA/swift.git
 cd swift
-git checkout ss-release-2.27.0.2
+export SWIFT_TAG="$( git describe --tags --abbrev=0 $(git rev-list --tags --max-count=1) )"
+git checkout $SWIFT_TAG
```
Comment (edmc-ss) [inline saio/vagrant_provision.sh:232]: Ditto.

File: sait/vagrant_provision.sh [hunk 4]
```diff
@@ -230,7 +230,8 @@ echo "export ST_KEY=testing" >> ~vagrant/.bash_profile
 cd ~swift
 git clone https://github.com/NVIDIA/swift.git
 cd swift
-git checkout ss-release-2.27.0.2
+export SWIFT_TAG="$( git describe --tags --abbrev=0 $(git rev-list --tags --max-count=1) )"
```
Comment (edmc-ss) [inline sait/vagrant_provision.sh:233]: Ditto.

**Review by edmc-ss — CHANGES_REQUESTED** [review]
Basically, I think the derivation of the SwiftVersion should be done *OUTSIDE* of the Dockerfile/provitioning script... and, in so doing, affect the "tag" applied to the created Docker Image.

**Comment by gerardgine** [comment]
As discussed in real life, all your points are perfectly valid, but they're also outside the scope of what we're trying to achieve in this PR.

Merging!

**Pull request merged**